modules/container/set.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package container
+
+type Set[T comparable] map[T]struct{}
+
+// NewSet creates a set and adds the specified elements to it.
+func NewSet[T comparable](values ...T) Set[T] {
+	s := make(Set[T], len(values))
+	s.AddMultiple(values...)
+	return s
+}
+
+// Add adds the specified element to a set.
+// Returns true if the element is added; false if the element is already present.
+func (s Set[T]) Add(value T) bool {
+	if _, has := s[value]; !has {
+		s[value] = struct{}{}
+		return true
+	}
+	return false
+}
+
+// AddMultiple adds the specified elements to a set.
+func (s Set[T]) AddMultiple(values ...T) {
```

Comment:
Why not just use Add to also add multiple? Just like you can use append adding single or multiple values

Reply:
But it's ambiguous when reading code `Add(a,b)` to set `[a]`, only `b` is added. What if one day there is another function `Add` with comment "true if every element is added". The return value for adding multiple values is meaningless

I stand the current approach.

Reply:
That's discutable but I won't block. For our use case single Add would be more than enough and easier to use imho

Reply:
Yup, different frameworks have different choices. For C++ stl, `pair<iterator,bool> insert(elem)`  vs `void insert(begin, end)` (like current approach), while in Java `bool add(elem)` vs `bool addAll(elemList)` (return true if the set is changed).

Personally I like the current approach 😁